src/python/pants/engine/process.py

```diff
@@ -108,6 +108,8 @@ def __init__(
             result = await Get(ProcessResult, Process(["/bin/echo", "hello world"], description="demo"))
             assert result.stdout == b"hello world"
         """
+        if isinstance(argv, str):
```

Comment:
This is a bit of a path to madness. If we're not fully typechecked I  guess this makes sense but otherwise it begs the question of why not runtime check the types of all the other constructor params.

Reply:
So, yes... but that's also a slippery slope argument. I'd posit that there exist some common usecases where it will make sense to have this sort of defense... and others where it just won't.

In the case of `argv`, this defends against a very common mistake that I happened to make while developing this change.